### ISSUE TITLE
Home-Assistant not recognizing successful action

### ISSUE THREAD

**Issue by maretodoric** [mattsaxon/sonoff-lan-mode-homeassistant#29]
First of all, great project!

I have a custom installation of HA on CentOS 7 in Python virtual environment. I've added this module to HA and it successfully loaded it. However, there is a specific following issue happening.

Switch initial state is OFF:
I try to turn on the switch from HA, the switch WILL turn itself ON.
Switch will (supposedly) send update status which is received by HA but that status is not updated, maybe not recognized? The "On/Off" slider on HA web page will revert to off and following log entry is noted after enabling debug log and after successful switch "turn on" action:
```
Apr 13 19:13:40 homeassistant hass: 2020-04-13 19:13:40 DEBUG (SyncWorker_4) [custom_components.sonoff_lan_mode_r3.switch] response received: <Response [200]> b'{"seq":2,"sequence":"1586798020116","error":0}'
```
200 message should indicate success, however, HA doesn't seem to recognize it.
Periodically, following will appear in log:
```Apr 13 19:20:40 homeassistant hass: 2020-04-13 19:20:40 WARNING (MainThread) [custom_components.sonoff_lan_mode_r3.switch] Device: 100082e44f. Update message not received in timeout period, retry
Apr 13 19:20:40 homeassistant hass: 2020-04-13 19:20:40 DEBUG (MainThread) [custom_components.sonoff_lan_mode_r3.switch] send_updated_params_loop now awaiting event
Apr 13 19:20:40 homeassistant hass: 2020-04-13 19:20:40 DEBUG (MainThread) [custom_components.sonoff_lan_mode_r3.switch] Connected!
Apr 13 19:20:40 homeassistant hass: 2020-04-13 19:20:40 DEBUG (MainThread) [custom_components.sonoff_lan_mode_r3.switch] params: {'switch': 'on'}
Apr 13 19:20:40 homeassistant hass: 2020-04-13 19:20:40 DEBUG (MainThread) [custom_components.sonoff_lan_mode_r3.switch] encrypted: {'sequence': '1586798440691', 'deviceid': '100082e44f', 'selfApikey': '123', 'iv': 'jQdNdlUcr+hOwGf4AcJO6Q==', 'encrypt': True, 'data': '9LobXxiu43PZ9dt9l7p2JQ=='}
Apr 13 19:20:40 homeassistant hass: 2020-04-13 19:20:40 DEBUG (SyncWorker_1) [custom_components.sonoff_lan_mode_r3.switch] Sending http message to http://192.168.0.22:8081/zeroconf/switch: {"sequence":"1586798440691","deviceid":"100082e44f","selfApikey":"123","iv":"jQdNdlUcr+hOwGf4AcJO6Q==","encrypt":true,"data":"9LobXxiu43PZ9dt9l7p2JQ=="}
Apr 13 19:20:40 homeassistant hass: 2020-04-13 19:20:40 DEBUG (MainThread) [custom_components.sonoff_lan_mode_r3.switch] Connection issue for device 100082e44f: HTTPConnectionPool(host='192.168.0.22', port=8081): Max retries exceeded with url: /zeroconf/switch (Caused by ProtocolError('Connection aborted.', ConnectionResetError(104, 'Connection reset by peer')))
```
...and looping
If i try to turn off the switch from eWeLink app BEFORE "Max retries" occurs, device will receive On request from HA again but still not update it's status on HA page.

I don't know why is this happening, however the switch is not standard Sonoff switch, so maybe that's the reason. It is based on 3.x firmware and manufactured by coolkit.

Bellow is my entry in configuration.yml
```
switch:
  - platform: sonoff_lan_mode_r3
    name: Ventil - Dvoriste
    device_id: 100082e44f
    api_key: ffffffff-ffff-ffff-ffff-ffffffffffff # not needed for devices in DIY mode
    icon: mdi:lightbulb
    outlet: 0
```

I captured apikey by connecting laptop to WiFi of switch during pairing mode and then captured the handshake using wireshark while pairing the device from eWeLink app from another device, so apikey is supposed to be ok. Well, then again, if it wasn't correct, switch would not even turn on the first time.

If there is ANYTHING i can do to help troubleshoot this further on , please let me know ! I would do whatever is needed.

**Comment by mattsaxon** [comment]
So looks like you've everything setup correctly.

The logs you have are 7 mins apart, would be good to see them around the failure.

The second set of logs suggest that the endpoint is not responding, but clearly in your first log it has responded.

The way that the device should work is that when the switch changes state, an mDNS message is sent which changes the slide in HA, look like that isn't happening.

If you are able to do it, using pysonofflanr3 (this is the library that this component calls) in debug mode to switch the device would be useful, this then takes all the HA stuff away and I'll be able to see a nice clean debug log.

If you can't get this working, just send me your full debug log.

**Comment by maretodoric** [comment]
Hey Matt,

looks like there was an issue with my HA installation. I've performed a manual install and and using Python 3.6. I've reinstalled it by using docker which is using Python 3.7, so that _might_ be the case... Maybe... Don't know. Works fine since I reinstalled HA anyway! :) thanks! This is great project!
I have just obtained more Sonoff switches, added them as well and everything is working fine!
I have another bug but that appears to be the issue with ewelink/sonoff firmware, if device doesn't have internet access, it will re-connect to WiFi every one minute, basically spamming with DHCP requests if multiple switches are in network. It pisses me off, i will have to contact eWeLink again to report this, since reporting to seller clearly doesn't do anything..

On a side note, i would like to add that non sonoff brand switches but based on eWeLink firmware work with this as well! And without issue!
I've tested three of these: https://www.aliexpress.com/item/4000274226640.html, works like a charm with sonoff_lan_mode_r3 setup.
Once again, GREAT project!
